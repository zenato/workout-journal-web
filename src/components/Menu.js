import React from 'react';
import { NavLink } from 'react-router-dom';
import Cookies from 'js-cookie';
import classNames from 'classnames/bind';
import styles from './Menu.scss';

const cx = classNames.bind(styles);

const logout = (e) => {
  e.preventDefault();
  Cookies.remove('accessToken');
  window.location.href = '/';
};

const Menu = ({ user }) => {
  return (
    <div className={cx('menu')}>
      <ul>
        <li><NavLink exact to="/" activeClassName={cx('active')}>Home</NavLink></li>
        <li><NavLink to="/events" activeClassName={cx('active')}>Events</NavLink></li>
        <li><NavLink to="/posts" activeClassName={cx('active')}>Posts</NavLink></li>
        {user && (
          <li>
            <a href="/logout" onClick={logout}>Logout</a>
          </li>
        )}
      </ul>
    </div>
  );
};

export default Menu;
