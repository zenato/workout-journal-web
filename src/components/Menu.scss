@import 'common';

.menu {
  $color: #f2f2f2;

  background-color: #323232;
  color: $color;
  padding: 10px;
  maring: 0;
  font-size: 1.2rem;

  a {
    color: $color;
    &:hover, &:active, &:visited {
      color: $color;
    }
  }

  li:not(:last-child) {
    float: left;
    margin-right: 10px;
  }

  .active {
    color: red !important;
    font-weight: bold;
  }
}
