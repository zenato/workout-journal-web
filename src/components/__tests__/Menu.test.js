import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../Menu';


it('renders without crashing', () => {
  const wrapper = shallow(<Menu />);
  expect(wrapper).toMatchSnapshot();
});
