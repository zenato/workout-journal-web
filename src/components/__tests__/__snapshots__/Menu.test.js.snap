// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="menu"
>
    <ul>
        <li>
            <NavLink
                activeClassName="active"
                exact={true}
                to="/"
            >
                Home
            </NavLink>
        </li>
        <li>
            <NavLink
                activeClassName="active"
                to="/events"
            >
                Events
            </NavLink>
        </li>
        <li>
            <NavLink
                activeClassName="active"
                to="/posts"
            >
                Posts
            </NavLink>
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div
      className="menu"
>
      <ul>
            <li>
                  <NavLink
                        activeClassName="active"
                        exact={true}
                        to="/"
                  >
                        Home
                  </NavLink>
            </li>
            <li>
                  <NavLink
                        activeClassName="active"
                        to="/events"
                  >
                        Events
                  </NavLink>
            </li>
            <li>
                  <NavLink
                        activeClassName="active"
                        to="/posts"
                  >
                        Posts
                  </NavLink>
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Menu />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="menu"
>
          <ul>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        exact={true}
                                        to="/"
                              >
                                        Home
                              </NavLink>
                    </li>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        to="/events"
                              >
                                        Events
                              </NavLink>
                    </li>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        to="/posts"
                              >
                                        Posts
                              </NavLink>
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="menu"
>
          <ul>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        exact={true}
                                        to="/"
                              >
                                        Home
                              </NavLink>
                    </li>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        to="/events"
                              >
                                        Events
                              </NavLink>
                    </li>
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        to="/posts"
                              >
                                        Posts
                              </NavLink>
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Menu />,
}
`;
